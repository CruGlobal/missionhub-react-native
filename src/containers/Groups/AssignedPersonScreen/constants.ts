export const CONTACT_PERSON_SCREEN = 'nav/CONTACT_PERSON';
export const IS_USER_CREATED_MEMBER_PERSON_SCREEN =
  'nav/IS_USER_CREATED_MEMBER_PERSON';
export const IS_GROUPS_MEMBER_PERSON_SCREEN = 'nav/IS_GROUPS_MEMBER_PERSON';
export const MEMBER_PERSON_SCREEN = 'nav/MEMBER_PERSON';
export const ME_PERSONAL_PERSON_SCREEN = 'nav/ME_PERSONAL_PERSON';
export const IS_GROUPS_ME_COMMUNITY_PERSON_SCREEN =
  'nav/IS_GROUPS_ME_COMMUNITY_PERSON';
export const ME_COMMUNITY_PERSON_SCREEN = 'nav/ME_COMMUNITY_PERSON';
