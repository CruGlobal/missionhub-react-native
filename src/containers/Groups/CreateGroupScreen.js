import React, { Component } from 'react';
import { connect } from 'react-redux';
import { translate } from 'react-i18next';

import { Flex, Text } from '../../components/common';
import Header from '../Header';
import BackButton from '../BackButton';

@translate('groupsCreateGroup')
class CreateGroupScreen extends Component {
  render() {
    return (
      <Flex value={1}>
        <Header left={<BackButton />} title="Create Group" />
        <Text>Create Group</Text>
      </Flex>
    );
  }
}

CreateGroupScreen.propTypes = {};

const mapStateToProps = (state, { navigation }) => ({
  ...(navigation.state.params || {}),
});

export default connect(mapStateToProps)(CreateGroupScreen);
export const CREATE_GROUP_SCREEN = 'nav/CREATE_GROUP_SCREEN';
