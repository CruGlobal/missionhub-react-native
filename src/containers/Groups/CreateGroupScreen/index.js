import React, { Component } from 'react';
import {
  ScrollView,
  View,
  Keyboard,
  Image,
  KeyboardAvoidingView,
} from 'react-native';
import { connect } from 'react-redux';
import { translate } from 'react-i18next';

import {
  Flex,
  Text,
  Input,
  IconButton,
  Button,
} from '../../../components/common';
import Header from '../../Header';
import theme from '../../../theme';
import CAMERA_ICON from '../../../../assets/images/cameraIcon.png';
import { navigateBack } from '../../../actions/navigation';
import ImagePicker from '../../../components/ImagePicker';

import styles from './styles';

@translate('groupsCreateGroup')
class CreateGroupScreen extends Component {
  state = {
    name: '',
    imageUri: null,
  };

  onChangeText = text => this.setState({ name: text });

  createCommunity = () => {
    Keyboard.dismiss();
    const text = (this.state.name || '').trim();
    if (!text) {
      return;
    }

    // TODO: Create community
    // console.log('create community', text);
  };

  handleImageChange = data => this.setState({ imageUri: data.uri });

  navigateBack = () => this.props.dispatch(navigateBack());

  ref = c => (this.nameInput = c);

  renderImage() {
    const { imageUri } = this.state;
    if (imageUri) {
      return (
        <Image
          resizeMode="cover"
          source={{ uri: imageUri }}
          style={styles.image}
        />
      );
    }
    return <Image source={CAMERA_ICON} />;
  }

  render() {
    const { t } = this.props;

    return (
      <View style={styles.container}>
        <Header
          left={
            <IconButton
              name="deleteIcon"
              type="MissionHub"
              onPress={this.navigateBack}
            />
          }
          shadow={false}
          title={t('createCommunity')}
        />
        <ScrollView keyboardShouldPersistTaps="handled" style={styles.flex}>
          <ImagePicker onSelectImage={this.handleImageChange}>
            <Flex align="center" justify="center" style={styles.imageWrap}>
              {this.renderImage()}
            </Flex>
          </ImagePicker>

          <KeyboardAvoidingView
            keyboardVerticalOffset={theme.buttonHeight}
            style={styles.flex}
          >
            <Flex style={styles.fieldWrap}>
              <Text style={styles.label} type="header">
                {t('name')}
              </Text>
              <Input
                ref={this.ref}
                onChangeText={this.onChangeText}
                value={this.state.name}
                autoFocus={true}
                autoCorrect={true}
                selectionColor={theme.white}
                returnKeyType="done"
                style={styles.input}
                blurOnSubmit={true}
                placeholder=""
              />
            </Flex>
          </KeyboardAvoidingView>
        </ScrollView>

        <Flex align="stretch" justify="end">
          <Button
            type="secondary"
            onPress={this.createCommunity}
            text={t('createCommunity').toUpperCase()}
            style={styles.createButton}
          />
        </Flex>
      </View>
    );
  }
}

CreateGroupScreen.propTypes = {};

const mapStateToProps = (state, { navigation }) => ({
  ...(navigation.state.params || {}),
});

export default connect(mapStateToProps)(CreateGroupScreen);
export const CREATE_GROUP_SCREEN = 'nav/CREATE_GROUP_SCREEN';
