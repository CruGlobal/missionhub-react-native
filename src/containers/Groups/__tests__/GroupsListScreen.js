import React from 'react';
import configureStore from 'redux-mock-store';

import GroupsListScreen from '../GroupsListScreen';
import { renderShallow } from '../../../../testUtils';
import { navigatePush } from '../../../actions/navigation';
import { getMyCommunities } from '../../../actions/organizations';
import { trackActionWithoutData } from '../../../actions/analytics';
import { communitiesSelector } from '../../../selectors/organizations';
import * as common from '../../../utils/common';
import { GROUP_SCREEN, USER_CREATED_GROUP_SCREEN } from '../GroupScreen';
import { CREATE_GROUP_SCREEN } from '../CreateGroupScreen';
import { checkForUnreadComments } from '../../../actions/unreadComments';
import { resetScrollGroups } from '../../../actions/swipe';
import { ACTIONS } from '../../../constants';
import {
  CREATE_COMMUNITY_UNAUTHENTICATED_FLOW,
  JOIN_BY_CODE_FLOW,
} from '../../../routes/constants';

jest.mock('../../../selectors/organizations');
jest.mock('../../../actions/unreadComments');
jest.mock('../../../actions/navigation', () => ({
  navigatePush: jest.fn(() => ({ type: 'test' })),
}));
jest.mock('../../../actions/organizations', () => ({
  getMyCommunities: jest.fn(() => ({ type: 'test' })),
}));
jest.mock('../../../actions/swipe', () => ({
  resetScrollGroups: jest.fn(() => ({ type: 'reset' })),
}));
jest.mock('../../../actions/analytics');
jest.mock('../../TrackTabChange', () => () => null);

const mockStore = configureStore();
const organizations = {
  all: [
    {
      id: '1',
      name: 'Test Org 1',
    },
    {
      id: '2',
      name: 'Test Org 2',
      user_created: true,
    },
  ],
};
const auth = { isFirstTime: false };
const swipe = {};
const store = mockStore({ organizations, auth, swipe });

beforeEach(() => {
  navigatePush.mockReturnValue({ type: 'test' });
  getMyCommunities.mockReturnValue({ type: 'test' });
  trackActionWithoutData.mockReturnValue({ type: 'test' });
  communitiesSelector.mockReturnValue(organizations.all);
  checkForUnreadComments.mockReturnValue({
    type: 'check for unread comments',
  });
  common.refresh = jest.fn((_, refreshMethod) => refreshMethod());
});

it('should render null state', () => {
  const component = renderShallow(
    <GroupsListScreen />,
    mockStore({ organizations: { all: [] }, auth, swipe }),
  );
  expect(component).toMatchSnapshot();
});

describe('GroupsListScreen', () => {
  let component;
  beforeEach(() => {
    component = renderShallow(<GroupsListScreen />, store);
  });

  it('should render correctly', () => {
    expect(component).toMatchSnapshot();
  });

  describe('handlePress', () => {
    it('navigates to groups screen', () => {
      const organization = organizations.all[0];
      const item = component
        .childAt(3)
        .childAt(0)
        .props()
        .renderItem({ item: organization });
      item.props.onPress(organization);

      expect(navigatePush).toHaveBeenCalledWith(GROUP_SCREEN, { organization });
      expect(communitiesSelector).toHaveBeenCalledWith({
        organizations,
        auth,
      });
      expect(trackActionWithoutData).toHaveBeenCalledWith(
        ACTIONS.SELECT_COMMUNITY,
      );
    });

    it('navigates to user created org screen', () => {
      const organization = organizations.all[1];
      const item = component
        .childAt(3)
        .childAt(0)
        .props()
        .renderItem({ item: organization });
      item.props.onPress(organization);

      expect(navigatePush).toHaveBeenCalledWith(USER_CREATED_GROUP_SCREEN, {
        organization,
      });
      expect(communitiesSelector).toHaveBeenCalledWith({
        organizations,
        auth,
      });
      expect(trackActionWithoutData).toHaveBeenCalledWith(
        ACTIONS.SELECT_COMMUNITY,
      );
    });
  });

  it('should open main menu', () => {
    common.openMainMenu = jest.fn(() => ({ type: 'test' }));
    const instance = component.instance();
    instance.openMainMenu();
    expect(common.openMainMenu).toHaveBeenCalled();
  });

  it('should render item', () => {
    const instance = component.instance();
    const renderedItem = instance.renderItem({
      item: { id: '1', name: 'test' },
    });
    expect(renderedItem).toMatchSnapshot();
  });

  it('should load groups', () => {
    const instance = component.instance();
    instance.loadGroups();
    expect(getMyCommunities).toHaveBeenCalled();
  });

  it('should load groups on mount', async () => {
    const instance = component.instance();
    instance.loadGroups = jest.fn();
    await instance.componentDidMount();
    expect(instance.loadGroups).toHaveBeenCalled();
  });

  it('should load groups and scroll to index 0', async () => {
    const store = mockStore({
      organizations,
      auth,
      swipe: { groupScrollToId: '1' },
    });
    component = renderShallow(<GroupsListScreen />, store);
    const instance = component.instance();
    instance.flatList = { scrollToIndex: jest.fn() };
    instance.loadGroups = jest.fn();
    await instance.componentDidMount();
    expect(instance.loadGroups).toHaveBeenCalled();
    expect(resetScrollGroups).toHaveBeenCalled();
    expect(instance.flatList.scrollToIndex).toHaveBeenCalledWith({
      animated: true,
      index: 0,
      viewPosition: 0,
    });
  });

  it('should load groups and scroll to index 1', async () => {
    const store = mockStore({
      organizations,
      auth,
      swipe: { groupScrollToId: '2' },
    });
    component = renderShallow(<GroupsListScreen />, store);
    const instance = component.instance();
    instance.flatList = { scrollToIndex: jest.fn() };
    instance.loadGroups = jest.fn();
    await instance.componentDidMount();
    expect(instance.loadGroups).toHaveBeenCalled();
    expect(resetScrollGroups).toHaveBeenCalled();
    expect(instance.flatList.scrollToIndex).toHaveBeenCalledWith({
      animated: true,
      index: 1,
      viewPosition: 0.5,
    });
  });

  it('should load groups and not scroll to index', async () => {
    const store = mockStore({
      organizations,
      auth,
      swipe: { groupScrollToId: 'doesnt exist' },
    });
    component = renderShallow(<GroupsListScreen />, store);
    const instance = component.instance();
    instance.flatList = { scrollToIndex: jest.fn() };
    instance.loadGroups = jest.fn();
    await instance.componentDidMount();
    expect(instance.loadGroups).toHaveBeenCalled();
    expect(instance.flatList.scrollToIndex).toHaveBeenCalledTimes(0);
    expect(resetScrollGroups).toHaveBeenCalled();
  });

  it('should refresh the list', () => {
    const instance = component.instance();

    instance.handleRefresh();

    expect(checkForUnreadComments).toHaveBeenCalled();
    expect(common.refresh).toHaveBeenCalledWith(instance, instance.loadGroups);
    expect(getMyCommunities).toHaveBeenCalled();
  });

  it('should render null', () => {
    const instance = component.instance();
    const renderedItem = instance.renderNull();
    expect(renderedItem).toMatchSnapshot();
  });

  it('navigates to join group screen', () => {
    component
      .childAt(2)
      .childAt(0)
      .childAt(0)
      .props()
      .onPress();

    expect(navigatePush).toHaveBeenCalledWith(JOIN_BY_CODE_FLOW);
  });

  it('navigates to create group screen', () => {
    component
      .childAt(2)
      .childAt(1)
      .childAt(0)
      .props()
      .onPress();

    expect(navigatePush).toHaveBeenCalledWith(CREATE_GROUP_SCREEN);
  });

  it('navigates to Upgrade Account Screen if not signed in', () => {
    const store = mockStore({
      organizations,
      auth: { isFirstTime: true },
      swipe,
    });

    component = renderShallow(<GroupsListScreen />, store);

    component
      .childAt(2)
      .childAt(1)
      .childAt(0)
      .props()
      .onPress();

    expect(navigatePush).toHaveBeenCalledWith(
      CREATE_COMMUNITY_UNAUTHENTICATED_FLOW,
    );
  });
});
