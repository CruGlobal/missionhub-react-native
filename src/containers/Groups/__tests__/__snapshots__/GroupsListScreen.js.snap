// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupsListScreen should render correctly 1`] = `
<SafeAreaView
  style={
    Object {
      "backgroundColor": "#007398",
      "flex": 1,
    }
  }
>
  <Component
    screen="GroupsTab"
  />
  <Header
    left={
      <IconButton
        name="menuIcon"
        onPress={[Function]}
        testID="IconButton"
        type="MissionHub"
      />
    }
    title="COMMUNITIES"
  />
  <Flex
    direction="row"
  >
    <Flex
      value={1}
    >
      <Button
        buttonTextStyle={
          Object {
            "color": "#ffffff",
            "fontSize": 12,
            "fontWeight": "bold",
          }
        }
        onPress={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "alignSelf": "stretch",
              "backgroundColor": "#005A7F",
              "flexBasis": 40,
              "justifyContent": "center",
              "paddingVertical": 13,
            },
            Object {
              "borderRightColor": "#007398",
              "borderRightWidth": 1,
            },
          ]
        }
        text="JOIN A COMMUNITY"
        type="transparent"
      />
    </Flex>
    <Flex
      value={1}
    >
      <Button
        buttonTextStyle={
          Object {
            "color": "#ffffff",
            "fontSize": 12,
            "fontWeight": "bold",
          }
        }
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "backgroundColor": "#005A7F",
            "flexBasis": 40,
            "justifyContent": "center",
            "paddingVertical": 13,
          }
        }
        text="CREATE A COMMUNITY"
        type="transparent"
      />
    </Flex>
  </Flex>
  <ScrollViewMock
    refreshControl={
      <MyRefreshControl
        onRefresh={[Function]}
        refreshing={false}
      />
    }
    style={
      Object {
        "backgroundColor": "#B4B6BA",
        "flex": 1,
      }
    }
  >
    <FlatList
      data={
        Array [
          Object {
            "id": "1",
            "name": "Test Org 1",
          },
          Object {
            "id": "2",
            "name": "Test Org 2",
            "user_created": true,
          },
        ]
      }
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      style={
        Object {
          "backgroundColor": "#ECEEF2",
          "flex": 1,
        }
      }
      testID="FlatList"
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
  </ScrollViewMock>
</SafeAreaView>
`;

exports[`should render null state 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#007398",
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "height": 56,
          "paddingTop": 0,
        },
        undefined,
        null,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "paddingLeft": 5,
          },
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "justifyContent": "center",
          },
        ]
      }
    >
      <View
        accessible={true}
        clickable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="IconButton"
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              null,
              Object {},
              null,
            ]
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Array [
                  Object {
                    "color": "#ffffff",
                  },
                  Object {
                    "fontSize": 18,
                  },
                  Array [
                    Object {
                      "margin": 10,
                    },
                    Object {},
                  ],
                ],
                Object {
                  "fontFamily": "icomoon",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            î¤š
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {},
          Object {
            "alignItems": "center",
            "flex": 4,
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#505256",
            },
            Object {
              "fontFamily": "SourceSansPro-Regular",
            },
            Array [
              Object {
                "color": "white",
                "fontSize": 14,
                "fontWeight": "500",
              },
              undefined,
            ],
          ]
        }
      >
        COMMUNITIES
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "paddingRight": 5,
          },
          Object {
            "alignItems": "flex-end",
            "flex": 1,
            "justifyContent": "center",
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {},
        Object {
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {},
          Object {
            "flex": 1,
          },
        ]
      }
    >
      <View
        accessible={true}
        clickable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              null,
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "stretch",
                  "backgroundColor": "#005A7F",
                  "flexBasis": 40,
                  "justifyContent": "center",
                  "paddingVertical": 13,
                },
                Object {
                  "borderRightColor": "#007398",
                  "borderRightWidth": 1,
                },
              ],
              null,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#505256",
                },
                Object {
                  "fontFamily": "SourceSansPro-Regular",
                },
                Array [
                  Object {
                    "backgroundColor": "rgba(0,0,0,0)",
                    "color": "#ffffff",
                    "fontSize": 18,
                    "fontWeight": "500",
                    "letterSpacing": 1.5,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#ffffff",
                    "fontSize": 12,
                    "fontWeight": "bold",
                  },
                ],
              ]
            }
          >
            JOIN A COMMUNITY
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {},
          Object {
            "flex": 1,
          },
        ]
      }
    >
      <View
        accessible={true}
        clickable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              null,
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "backgroundColor": "#005A7F",
                "flexBasis": 40,
                "justifyContent": "center",
                "paddingVertical": 13,
              },
              null,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#505256",
                },
                Object {
                  "fontFamily": "SourceSansPro-Regular",
                },
                Array [
                  Object {
                    "backgroundColor": "rgba(0,0,0,0)",
                    "color": "#ffffff",
                    "fontSize": 18,
                    "fontWeight": "500",
                    "letterSpacing": 1.5,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#ffffff",
                    "fontSize": 12,
                    "fontWeight": "bold",
                  },
                ],
              ]
            }
          >
            CREATE A COMMUNITY
          </Text>
        </View>
      </View>
    </View>
  </View>
  <RCTScrollView
    refreshControl={
      <MyRefreshControl
        onRefresh={[Function]}
        refreshing={false}
      />
    }
    style={
      Object {
        "backgroundColor": "#B4B6BA",
        "flex": 1,
      }
    }
  >
    <RCTRefreshControl />
    <View>
      <RCTScrollView
        data={
          Array [
            Object {
              "id": "1",
              "name": "Test Org 1",
            },
            Object {
              "id": "2",
              "name": "Test Org 2",
              "user_created": true,
            },
          ]
        }
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        numColumns={1}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        style={
          Object {
            "backgroundColor": "#ECEEF2",
            "flex": 1,
          }
        }
        testID="FlatList"
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              accessible={true}
              clickable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ffffff",
                  "borderRadius": 4,
                  "elevation": 4,
                  "flexDirection": "row",
                  "marginHorizontal": 16,
                  "marginVertical": 8,
                  "opacity": 1,
                  "shadowColor": "#000000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.3,
                  "shadowRadius": 2,
                }
              }
            >
              <View
                style={
                  Array [
                    Array [
                      Object {
                        "borderRadius": 4,
                        "height": 403.875,
                      },
                      undefined,
                    ],
                    Object {
                      "flex": 1,
                    },
                  ]
                }
              >
                <Image
                  resizeMode="cover"
                  source={
                    Object {
                      "testUri": "../../../assets/images/impactBackground.png",
                    }
                  }
                  style={
                    Object {
                      "borderRadius": 4,
                      "height": "100%",
                      "width": "100%",
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "rgba(0, 0, 0, 0.6)",
                        "borderBottomLeftRadius": 4,
                        "borderBottomRightRadius": 4,
                        "bottom": 0,
                        "left": 0,
                        "padding": 13,
                        "position": "absolute",
                        "right": 0,
                      },
                      Object {
                        "flexDirection": "row",
                        "justifyContent": "center",
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {},
                        Object {
                          "flex": 1,
                        },
                      ]
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#505256",
                          },
                          Object {
                            "fontFamily": "SourceSansPro-Regular",
                          },
                          Object {
                            "color": "#ffffff",
                            "fontSize": 14,
                            "fontWeight": "bold",
                          },
                        ]
                      }
                    >
                      TEST ORG 1
                    </Text>
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#505256",
                          },
                          Object {
                            "fontFamily": "SourceSansPro-Regular",
                          },
                          Object {
                            "color": "#ffffff",
                            "fontSize": 12,
                          },
                        ]
                      }
                    >
                      0 Contacts
                      <Text
                        style={
                          Array [
                            Object {
                              "color": "#505256",
                            },
                            Object {
                              "fontFamily": "SourceSansPro-Regular",
                            },
                            Object {},
                          ]
                        }
                      >
                          Â·  
                      </Text>
                      0 Unassigned
                    </Text>
                  </View>
                  <View
                    style={
                      Array [
                        Object {},
                        Object {
                          "alignItems": "center",
                          "justifyContent": "center",
                        },
                      ]
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      style={
                        Array [
                          Object {
                            "color": undefined,
                            "fontSize": 12,
                          },
                          Array [
                            Object {
                              "color": "#ffffff",
                            },
                            Object {
                              "fontSize": 20,
                            },
                            Object {
                              "color": "#ffffff",
                            },
                          ],
                          Object {
                            "fontFamily": "icomoon",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      î¤„
                    </Text>
                    <View
                      style={
                        Object {
                          "backgroundColor": "#FF5532",
                          "borderColor": "#260C06",
                          "borderRadius": 8,
                          "borderWidth": 2,
                          "height": 16,
                          "position": "absolute",
                          "right": -5,
                          "top": 0,
                          "width": 16,
                        }
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              accessible={true}
              clickable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ffffff",
                  "borderRadius": 4,
                  "elevation": 4,
                  "flexDirection": "row",
                  "marginHorizontal": 16,
                  "marginVertical": 8,
                  "opacity": 1,
                  "shadowColor": "#000000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.3,
                  "shadowRadius": 2,
                }
              }
            >
              <View
                style={
                  Array [
                    Array [
                      Object {
                        "borderRadius": 4,
                        "height": 403.875,
                      },
                      Object {
                        "backgroundColor": "#005A7F",
                      },
                    ],
                    Object {
                      "flex": 1,
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "rgba(0, 0, 0, 0.6)",
                        "borderBottomLeftRadius": 4,
                        "borderBottomRightRadius": 4,
                        "bottom": 0,
                        "left": 0,
                        "padding": 13,
                        "position": "absolute",
                        "right": 0,
                      },
                      Object {
                        "flexDirection": "row",
                        "justifyContent": "center",
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {},
                        Object {
                          "flex": 1,
                        },
                      ]
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#505256",
                          },
                          Object {
                            "fontFamily": "SourceSansPro-Regular",
                          },
                          Object {
                            "color": "#ffffff",
                            "fontSize": 14,
                            "fontWeight": "bold",
                          },
                        ]
                      }
                    >
                      TEST ORG 2
                    </Text>
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#505256",
                          },
                          Object {
                            "fontFamily": "SourceSansPro-Regular",
                          },
                          Object {
                            "color": "#ffffff",
                            "fontSize": 12,
                          },
                        ]
                      }
                    >
                      0 Members
                    </Text>
                  </View>
                  <View
                    style={
                      Array [
                        Object {},
                        Object {
                          "alignItems": "center",
                          "justifyContent": "center",
                        },
                      ]
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      style={
                        Array [
                          Object {
                            "color": undefined,
                            "fontSize": 12,
                          },
                          Array [
                            Object {
                              "color": "#ffffff",
                            },
                            Object {
                              "fontSize": 20,
                            },
                            Object {
                              "color": "#ffffff",
                            },
                          ],
                          Object {
                            "fontFamily": "icomoon",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      î¤„
                    </Text>
                    <View
                      style={
                        Object {
                          "backgroundColor": "#FF5532",
                          "borderColor": "#260C06",
                          "borderRadius": 8,
                          "borderWidth": 2,
                          "height": 16,
                          "position": "absolute",
                          "right": -5,
                          "top": 0,
                          "width": 16,
                        }
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
